vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/pool/SqlConnectionPool.java

```diff
@@ -145,18 +154,32 @@ public int size() {
     return pool.size();
   }
 
-  public void evict() {
+  public void checkInvariants(long connectionTimeout) {
     long now = System.currentTimeMillis();
     pool.evict(conn -> conn.shouldEvict(now), ar -> {
       if (ar.succeeded()) {
         List<PooledConnection> res = ar.result();
         for (PooledConnection conn : res) {
           conn.close(Promise.promise());
         }
+        checkMin(connectionTimeout);
       }
     });
   }
 
+  public void checkMin(long connectionTimeout) {
+    if (pool.size() < minSize) {
+      ContextInternal context = vertx.getOrCreateContext();
```

Comment:
I think the issue here is that we might create connections with a context that is not best for the application and force context switching

Reply:
I assume you're referring to `vertx.getOrCreateContext()`. How would you solve this? Pass in a context?

Reply:
Hi, @tsegismont @vietj, any comments?